### ISSUE TITLE
False positive error "incompatible type" calling super() on a generic class with TypeVar constraints

### ISSUE THREAD

**Issue by leandro-lucarella-frequenz** [python/mypy#14774]
Reduced test case ([mypy-playground](https://mypy-play.net/?mypy=latest&python=3.11&flags=show-error-context%2Cshow-column-numbers%2Cshow-error-codes%2Cstrict%2Ccheck-untyped-defs%2Cdisallow-any-decorated%2Cdisallow-any-expr%2Cdisallow-any-explicit%2Cdisallow-any-generics%2Cdisallow-any-unimported%2Cdisallow-incomplete-defs%2Cdisallow-subclassing-any%2Cdisallow-untyped-calls%2Cdisallow-untyped-decorators%2Cdisallow-untyped-defs%2Cno-implicit-optional%2Cno-implicit-reexport%2Cno-strict-optional%2Cstrict-equality%2Cwarn-incomplete-stub%2Cwarn-redundant-casts%2Cwarn-return-any%2Cwarn-unreachable%2Cwarn-unused-configs%2Cwarn-unused-ignores&gist=0d988d1bb67e609cabd6521da88056ed)):

EDIT: Is not the same case actually, my case involves constraints.

```py
from typing import Generic, TypeVar

T = TypeVar("T", float, int)


class C(Generic[T]):
    def __init__(self, i: T) -> None:
        self.i: T = i


class B(C[T]):
    def __init__(self, i: T) -> None:
        super().__init__(i)
```

```
main.py: note: In member "__init__" of class "B":
main.py:13:26: error: Argument 1 to "__init__" of "C" has incompatible type "float"; expected "T"  [arg-type]
main.py:13:26: error: Argument 1 to "__init__" of "C" has incompatible type "int"; expected "T"  [arg-type]
Found 2 errors in 1 file (checked 1 source file)
```

_Originally posted by @leandro-lucarella-frequenz in https://github.com/python/mypy/issues/13566#issuecomment-1443582479_

**Comment by bjoernpollex-sc** [comment]
I'm running into the same issue. I've also noticed that, if the types in the constraint are related, then using a bound instead of constraint fixes the issue.
